scripts/test/shared.py

```diff
@@ -256,25 +256,31 @@ def setup_waterfall():
 # external tools
 
 try:
-  subprocess.check_call(
-      [NODEJS, '--version'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-except:
+  if NODEJS is not None:
+    subprocess.check_call(
+        [NODEJS, '--version'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+except OSError:
```

Comment:
Should this catch a `CalledProcessError` instead? (Likewise below)

Reply:
Alternatively, maybe they should all be unspecified `except Exception:` so we preserve the same semantics as before?

Reply:
Either way: I do like being more specific in what we catch and if there's an exception type that we don't expect, it would more or less serve the same purpose as an assert by letting it go unhandled.

Reply:
Right, clearly this case hid a NameError which had to be fixed.

Reply:
OK, I updated it to catch CalledProcessError, and to pass through others (e.g. NameError and TypeError thrown when typo or when NODEJS is None).